tests/tracking/test_tracking.py

```diff
@@ -100,16 +91,66 @@ def test_set_experiment():
         assert another_run.info.experiment_id == exp_id2.experiment_id
 
 
-def test_set_experiment_with_deleted_experiment_name():
+@pytest.mark.usefixtures("reset_active_experiment")
+def test_set_experiment_by_id():
+    name = "random_exp"
+    exp_id = mlflow.create_experiment(name)
+    mlflow.set_experiment(experiment_id=exp_id)
+    with start_run() as run:
+        assert run.info.experiment_id == exp_id
+
+    nonexistent_id = "-1337"
+    with pytest.raises(MlflowException) as exc:
+        mlflow.set_experiment(experiment_id=nonexistent_id)
+    assert exc.value.error_code == ErrorCode.Name(RESOURCE_DOES_NOT_EXIST)
+    with start_run() as run:
+        assert run.info.experiment_id == exp_id
+
+
+def test_set_experiment_parameter_validation():
+    with pytest.raises(MlflowException) as exc:
```

Comment:
Perhaps use the optional match= in `pytest.raises` to do regex matching for the exception error message?
```python
with pytest.raises.(MLflowException, match="Must specify exactly one") as exc:
    mlflow.set_experiment()
assert exc.value.error_Code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
```
Haru asked that I change a few newer unit tests to use this pattern.

Reply:
Awesome suggestion! Done!